osutil/user.go

```diff
@@ -150,12 +151,13 @@ func EnsureUserGroup(name string, id uint32, extraUsers bool) error {
 			delCmdStr = append(delCmdStr, "--extrausers")
 		}
 
-		// TODO: groupdel doesn't currently support --extrausers, so
-		// don't try to clean up when it is specified (LP: #1840375)
-		if !extraUsers {
-			delCmdStr = append(delCmdStr, name)
-			cmd = exec.Command(delCmdStr[0], delCmdStr[1:]...)
-			if output2, err2 := cmd.CombinedOutput(); err2 != nil {
+		delCmdStr = append(delCmdStr, name)
+		cmd = exec.Command(delCmdStr[0], delCmdStr[1:]...)
+		if output2, err2 := cmd.CombinedOutput(); err2 != nil {
+			if extraUsers && bytes.Contains(output2, []byte("unrecognized option")) {
+				// We have hit https://bugs.launchpad.net/bugs/1840375
+				return fmt.Errorf("groupdel does not support '--extrausers' option (after %s)", useraddErrStr)
```

Comment:
after reads a bit weird here, not sure the right way to phrase it is, but usually we will use the pattern `while doing x, error foo happened`, but really these two errors are orthogonal to each other IMHO, so maybe the thing to do here is to make them separate errors, one each line:

```
errors encountered ensuring group foo exists:
- groupdel does not support --extrausers option
- useradd failed with foo
```

Reply:
I've read about go's support for nested errors (with the `%w` modifier passed to `Errorf`); do you think it makes sense to use it here?

Reply:
As we discussed in the meeting, `%w` is only meant to be used in the case where callers want to inspect an error type, if we did something simple like:

```go
return fmt.Errorf("some syscall failed while doing foo: %v",err)
```

then if `err` was say a os.PathError, the standard library says you can check for that specific error with `os.IsNotExists`, but now we created a new error type from `Errorf` which doesn't match the os.PathError, so that check would fail. The solution from the Go team to this was to introduce `%w` which _wraps_ errors so that if we did this:

```
return fmt.Errorf("some syscall failed while doing foo: %w",err)
```

now the error returned by Errorf will be wrapped such that the os.IsNotExists function would return true since the IsNotExists function now can use the `Unwrap()` and `As()` or `Is()` functions from go's errors package (see https://golang.org/pkg/errors/). 

As was pointed out though, since we use an old go version for some distros, we cannot use the errors package directly, and instead we have to do sneaky tricks like use the xerrors package and define our format strings as constants, since old Go does not know about the `%w` verb. See here for an example: https://github.com/snapcore/snapd/blob/master/bootloader/lkenv/lkenv.go#L264-L275

Reply:
but also as explained wrapped errors is not what we want here since:
- the callers of this function probably don't care to inspect the different error cases
- the errors are not related to each other

Reply:
Done!